cpp/example_code/lambda/cpp_lambda/README.md

```diff
@@ -0,0 +1,116 @@
+# Example AWS Lambda function using the AWS C++ Lambda runtime
+
+Shows how to build and package an AWS Lambda function using the AWS C++ 
+Lambda runtime. This function can replace the Python Lambda function used in 
+the [Get started with functions](../get_started_with_functions_scenario.cpp) 
+scenario sample code.
+
+## Creating the Lamba function package 
+
+The Lambda function is built and packaged using [Docker](https://www.docker.com/).
+
+### Prerequisites
+
+[Docker](https://www.docker.com/) is needed to build and package the Lambda function. The Docker website contains installation instructions. 
+
+### Build and run the docker image
+
+This sample code contains a `DockerFile` which is used to build the Docker image. Using a shell application, navigate to the folder containing this README. 
+Then enter the following command:
+
+```bashcd 
+docker build . -f Dockerfile -t cpp_lambda_image
+```
+
+Now run the Docker image as a container using the following command:
+
+```bash
+docker run -i -t --name cpp_lambda --mount type=bind,source="$(pwd)",target=/cpp_lambda cpp_lambda_image bash
+```
+
+Your command line should now be running from within the Docker container. 
+The preceding run command should have bound the directory containing this 
+README to the Docker container.
+
+Test whether the host directory is correctly bound by entering the following 
+command:
+
+```bash
+ls /cpp_lambda
+```
+
+You should see the contents of this README's directory.
+
+```bash
+Dockerfile         README.md          calculator         increment   
+```
+
+### Build and package the Lambda function
+
+Build and install the [AWS Lambda C++ Runtime](https://github.com/awslabs/aws-lambda-cpp) from within the Docker container.
+
+```bash
+git clone https://github.com/awslabs/aws-lambda-cpp-runtime.git
+cd aws-lambda-cpp-runtime
+mkdir build
+cd build
+cmake3 .. -DCMAKE_BUILD_TYPE=Release \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DCMAKE_INSTALL_PREFIX=~/install 
+make
+make install
+```
+
+
+Now build the two Lambda functions used by the sample code. Because the Docker container is bound to the directory containing
+this README, the source code can be accessed within the container at `/cpp_lambda`.
+
+1. Build the "Increment" Lambda function code.
+
+```bash
+cd /cpp_lambda/increment
+mkdir build
+cd build
+cmake3 .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=~/install
+make
+make aws-lambda-package-cpp_lambda_calculator
+```
+
+2. Build the "Calculator" Lambda function code.
+
+```bash
+cd ../../calculator
+mkdir build
+cd build
+cmake3 .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=~/install
+make
+make aws-lambda-package-cpp_lambda_calculator
+```
+These build steps have generated two zip files which will be used by the sample 
+code in [Get started with functions](../get_started_with_functions_scenario.cpp).
+
+Copy the file `cpp_lambda/increment/build/cpp_lambda_calculator.zip` to the 
```

Comment:
Is the file in the increment directory also named cpp_lambda_calculator? I was a little confused here by the names vs the directories.

Reply:
This is a little difficult to explain. The executables must have the same name. This is referenced as the handler name. It is set in "CreateFunction", and can't be changed in "UpdateFunctionCode".  The way the lambda executable build process works, the executable name determines the zip file name. So two zip files with the same name are created (and the same executable name when unzipped), but with different code. One zip file is renamed before it is copied, so that both files can be in the same directory. I think this lambda is a deep-dive exercise, and hopefully the programmer can understand that with some effort.

Reply:
Alright, understood. Thanks for the explanation.